### PR TITLE
fix(dataplane) always set default envs

### LINKED ISSUE #245: env of deployment for dataplane could not be set when dataplane has env
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Similar to #63, The default env could not be set if dataplane has set `spec.deploymentoptions.env`. Kong uses `postgres` as its default database, so this will make Kong fail to start if env is set in `spec.deploymentoptions.env` , but `KONG_DATABASE` not set.

### Expected Behavior

When I specify env in dataplane, the default envs are still set.

### Steps To Reproduce

Similar to #63, create a dataplane like this:
```
apiVersion: gateway-operator.konghq.com/v1alpha1
kind: DataPlane
metadata:
  creationTimestamp: "2022-08-29T09:47:14Z"
  generation: 1
  name: dp1
spec:
  env:
  - name: TEST_ENV
    value: before_update

```
The default envs  are not set, and the deployment could not be started, because `KONG_DATABASE` is not set,.
### Kong Gateway Operator version

_No response_

### Kubernetes version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

Always set default envs of dataplane.

**Which issue this PR fixes**

fixes #245 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: internal/utils/dataplane/config.go [hunk 1]
```diff
@@ -73,12 +73,16 @@ var KongDefaults = map[string]string{
 // SetDataPlaneDefaults sets any unset default configuration options on the
 // DataPlane. No configuration is overridden. EnvVars are sorted
 // lexographically as a side effect.
-func SetDataPlaneDefaults(spec *operatorv1alpha1.DataPlaneDeploymentOptions) {
+// returns true if
```
Comment (pmalek) [inline internal/utils/dataplane/config.go:76]: This comments looks unfinished 🤔

File: test/integration/dataplane_test.go [hunk 2]
```diff
@@ -66,6 +77,22 @@ func TestDataplaneEssentials(t *testing.T) {
...
+	deployment := &deployments[0]
+
+	controllerContainer := k8sresources.GetPodContainerByName(
+		&deployment.Spec.Template.Spec, consts.DataPlaneProxyContainerName)
+	require.NotNil(t, controllerContainer)
+	envs := controllerContainer.Env
+	testEnvValue := getEnvValueByName(envs, "TEST_ENV")
+	require.Equal(t, "test", testEnvValue)
+	kongDatabaseEnvValue := getEnvValueByName(envs, consts.EnvVarKongDatabase)
+	require.Equal(t, "off", kongDatabaseEnvValue)
```
Comment (pmalek) [inline test/integration/dataplane_test.go:94]: This is coming from the default set of envs? Perhaps we comment that or add a message to the assertion?

File: internal/utils/dataplane/config.go (hunk 1 above)
Comment (randmonkey) [reply internal/utils/dataplane/config.go]: fixed.

File: test/integration/dataplane_test.go (hunk 2 above)
Comment (randmonkey) [reply test/integration/dataplane_test.go:96]: added comments here.

**Review by pmalek — APPROVED** [review]

**Pull request merged**